plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceSerializerModern.java

```diff
@@ -373,8 +373,14 @@ public boolean parseDataSources(
         var connectionConfigurationChanged = false;
 
         // Read in this particular order to handle configuration reading errors first, but process in reverse order later
+        Map<String, Map<String, Map<String, String>>>  secureCredentialsMap = null ;
         final var configurationMap = readConfiguration(configurationStorage, configurationManager, dataSourceIds);
-        final var secureCredentialsMap = readSecureCredentials(configurationStorage, configurationManager, dataSourceIds);
+        try {
+            secureCredentialsMap = readSecureCredentials(configurationStorage, configurationManager, dataSourceIds);
```

Comment:
This doesn't cover the case where the `readConfiguration` might fail as well.

Reply:
Generally, yes better to cover both functions